include/libcyphal/transport/can/can_transport.hpp

```diff
@@ -0,0 +1,141 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef LIBCYPHAL_TRANSPORT_CAN_TRANSPORT_HPP_INCLUDED
+#define LIBCYPHAL_TRANSPORT_CAN_TRANSPORT_HPP_INCLUDED
+
+#include "libcyphal/transport/can/media.hpp"
+#include "libcyphal/transport/errors.hpp"
+#include "libcyphal/transport/transport.hpp"
+
+#include <canard.h>
+#include <cetl/pf17/cetlpf.hpp>
+#include <cetl/pmr/function.hpp>
+
+#include <cstdint>
+
+namespace libcyphal
+{
+namespace transport
+{
+namespace can
+{
+
+/// @brief Defines interface of CAN transport layer.
+///
+class ICanTransport : public ITransport
+{
+public:
+    /// Defines structure for reporting transient transport errors to the user's handler.
+    ///
+    /// In addition to the error itself, it provides:
+    /// - Index of media interface related to this error.
+    ///   This index is the same as the index of the (not `nullptr`!) media interface
+    ///   pointer in the `media` span argument used at the `makeTransport()` factory method.
+    /// - A reference to the entity that has caused this error.
+    ///
+    struct TransientErrorReport
+    {
+        /// @brief Error report about pushing a message to a TX session.
+        struct CanardTxPush
+        {
+            AnyError        error;
+            std::uint8_t    media_index;
+            CanardInstance& culprit;
```

Comment:
Maybe we should make these pointers to ensure the error reports are copyable?

Reply:
Reports are copy constructible, but yes they are not copy **assign**able. IMHO it's ok limitation, especially b/c I didn't want to expose raw pointer interface (of `CanardInstance` or `IMedia`) - that is why I made them as `&` in the first place.